src/main/java/com/cairn/waypoint/dashboard/endpoints/HomeworkQuestionLinkedProtocolTemplate/LinkHomeworkQuestionAndProtocolTemplateEndpoint.java

```diff
@@ -0,0 +1,94 @@
+                                                           ProtocolTemplateRepository protocolTemplateRepository,
+                                                           HomeworkQuestionRepository homeworkQuestionRepository) {
+        this.linkedProtocolTemplateRepository = linkedProtocolTemplateRepository;
+        this.protocolTemplateRepository = protocolTemplateRepository;
+        this.homeworkQuestionRepository = homeworkQuestionRepository;
+    }
+
+    @PostMapping
+    @PreAuthorize("hasAnyAuthority('SCOPE_protocol.full', 'SCOPE_admin.full')")
+    @Operation(
+            summary = "Add a new HomeworkQuestionLinkedProtocolTemplate entry",
+            description = "Allows a user to link a homework question to a protocol template.",
+            security = @SecurityRequirement(name = "oAuth2JwtBearer"),
+            responses = {
+                    @ApiResponse(responseCode = "201", description = "Created - Homework question linked to protocol template successfully",
+                            content = {@Content(mediaType = "application/json",
+                                    schema = @Schema(implementation = HomeworkQuestionLinkedProtocolTemplate.class))}),
+                    @ApiResponse(responseCode = "400", description = "Bad Request",
+                            content = {@Content(schema = @Schema(hidden = true))}),
+                    @ApiResponse(responseCode = "404", description = "Not Found",
+                            content = {@Content(schema = @Schema(hidden = true))})
+            }
+
+    )
+    public ResponseEntity<?> addHomeworkQuestionLinkedProtocolTemplate(@RequestBody AddHomeworkQuestionLinkedProtocolTemplateDto dto) {
+        log.info("Linking homework question with ID [{}] to protocol template with ID [{}]",
+                dto.getHomeworkQuestionId(), dto.getProtocolTemplateId());
+
+        Optional<ProtocolTemplate> protocolTemplateOpt = protocolTemplateRepository.findById(dto.getProtocolTemplateId());
+        if (protocolTemplateOpt.isEmpty()) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND)
+                    .body("Protocol template with ID " + dto.getProtocolTemplateId() + " not found");
+        }
+
+        Optional<HomeworkQuestion> homeworkQuestionOpt = homeworkQuestionRepository.findById(dto.getHomeworkQuestionId());
+        if (homeworkQuestionOpt.isEmpty()) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND)
+                    .body("Homework question with ID " + dto.getHomeworkQuestionId() + " not found");
+        }
+
+        HomeworkQuestionLinkedProtocolTemplate linkedEntry = HomeworkQuestionLinkedProtocolTemplate.builder()
+                .protocolTemplate(protocolTemplateOpt.get())
+                .question(homeworkQuestionOpt.get())
+                .build();
+
+        HomeworkQuestionLinkedProtocolTemplate savedEntry = linkedProtocolTemplateRepository.save(linkedEntry);
```

Comment:
We should not be interacting with the Repository layer in the Controller. The entry to be saved should be passed via the Service Layer. Please add a service for the protocolTemplate (or update the existing) and create a method that does this. This keeps a layer of abstraction between the data layer and the controller layer